tracing-subscriber/src/filter/env/directive.rs

```diff
@@ -124,9 +125,11 @@ impl Directive {
             )
             .collect::<Result<FieldMap<_>, ()>>()
             .ok()?;
+        let target = meta.target();
         Some(field::CallsiteMatch {
             fields,
             level: self.level.clone(),
+            target: Some(Arc::from(target)),
```

Comment:
I don't think this is right --- you are using the target of the first callsite that matches the filter. Instead, you should use the target of the filter directive itself; the `target` field on `Directive`:
https://github.com/tokio-rs/tracing/blob/3fa215fd4a5c513b472ae7c55d2c85cd2c8e314f/tracing-subscriber/src/filter/env/directive.rs#L20
```suggestion
            target: self.target.clone(),
```

We should probably change `Directive`'s target field to be an `Arc`, and change the `s.to_owned()` here to `Arc::from(s)`: https://github.com/tokio-rs/tracing/blob/3fa215fd4a5c513b472ae7c55d2c85cd2c8e314f/tracing-subscriber/src/filter/env/directive.rs#L237